OpenRA.Mods.Common/Traits/World/WeatherOverlay.cs

```diff
@@ -0,0 +1,314 @@
+	{
+		[Desc("Factor for particle density. A smaller value increases the number of particles spawned.")]
+		public readonly int ParticleDensityFactor = 1311;
+
+		[Desc("The wind will change over time by the values of WindStrength, or just stick to the first value.")]
+		public readonly bool ChangingWind = true;
+
+		[Desc("Works only if ChangingWind is enabled. The levels of wind intensity desc. the x-movement value of the particles in px/tick.")]
+		public readonly int[] WindStrength = { -5, -3, -2, 0, 2, 3, 5 };
+
+		[Desc("Works only if ChangingWind is enabled. Min. and max. ticks needed to change the wind. ")]
+		public readonly int[] WindTick = { 150, 750 };
+
+		[Desc("The wind is fading softly from zero or instant.")]
+		public readonly bool InstantWindChanges = false;
+
+		[Desc("Particles are drawn with rectangles when enabled, else with lines.")]
+		public readonly bool UseRectangles = true;
+
+		[Desc("Works only with rectangles enabled. Size min. and max. value in pixels.")]
+		public readonly int[] PaticleSize = { 1, 3 };
+
+		[Desc("Scatters falling direction on the x-axis. Scatter min. and max. value in px/tick.")]
+		public readonly int[] ScatterDirection = { -1, 1 };
+
+		[Desc("The value that particles fall down. Gravity min. and max. value in px/tick.")]
+		public readonly float[] Gravity = { 1.00f, 2.00f };
+
+		[Desc("The current offset value for the swing movement. SwingOffset min. and max. value in px/tick.")]
+		public readonly float[] SwingOffset = { 1.0f, 1.5f };
+
+		[Desc("The value that particles swing to the side each update. SwingSpeed min. and max. value in px/tick.")]
+		public readonly float[] SwingSpeed = { 0.001f, 0.025f };
+
+		[Desc("The value range that can be swung to the left or right. SwingAmplitude min. and max. value in px/tick.")]
+		public readonly float[] SwingAmplitude = { 1.0f, 1.5f };
+
+		[Desc("The ArgbColors for the particles. Think about using the colors of the temperate.pal for example.")]
+		public readonly Color[] ParticleColors = {
+			Color.FromArgb(255, 236, 236, 236),
+			Color.FromArgb(255, 228, 228, 228),
+			Color.FromArgb(255, 208, 208, 208),
+			Color.FromArgb(255, 188, 188, 188)
+		};
+
+		[Desc("Works only with line enabled and will fade out the tail of the line.")]
+		public readonly byte LineTailAlphaValue = 200;
+
+		public object Create(ActorInitializer init) { return new WeatherOverlay(init.World, this); }
+	}
+
+	public class WeatherOverlay : ITick, IPostRender
+	{
+		readonly WeatherOverlayInfo info;
+		readonly World world;
+		struct Particle
+		{
+			public float PosX;
+			public float PosY;
+			public int Size;
+			public float DirectionScatterX;
+			public float Gravity;
+			public float SwingOffset;
+			public float SwingSpeed;
+			public int SwingDirection;
+			public float SwingAmplitude;
+			public Color Color;
+			public Color TailColor;
+		}
+
+		List<Particle> particleList = new List<Particle>();
+		int maxParticleCount;
```

Comment:
This value get created by the ParticleDensityFactorand and theScreensizes. 
To get it randomed would not make much sence at the current state in my opinion.
If we like to play with the particle count, it would make more sence to get a min and a max count and fade between them over a periode of time while the match is running. Like I already do it with the wind. That is something I'm aware of.